protocol/v2/blockchain/beacon/blind/blind.go

```diff
@@ -0,0 +1,285 @@
+				BLSToExecutionChanges:  p.Deneb.Block.Body.BLSToExecutionChanges,
+				BlobKZGCommitments:     p.Deneb.Block.Body.BlobKZGCommitments,
+			},
+		}
+
+		return &api.VersionedProposal{Version: p.Version, Blinded: true, DenebBlinded: bb}, bb, nil
+
+	case spec.DataVersionElectra:
+		if p.Electra == nil || p.Electra.Block == nil || p.Electra.Block.Body == nil || p.Electra.Block.Body.ExecutionPayload == nil {
+			return nil, nil, fmt.Errorf("electra block or payload is nil")
+		}
+		payload := p.Electra.Block.Body.ExecutionPayload
+		txRoot, err := (&utilbellatrix.ExecutionPayloadTransactions{Transactions: payload.Transactions}).HashTreeRoot()
+		if err != nil {
+			return nil, nil, fmt.Errorf("compute transactions root: %w", err)
+		}
+		wRoot, err := (&utilcapella.ExecutionPayloadWithdrawals{Withdrawals: payload.Withdrawals}).HashTreeRoot()
+		if err != nil {
+			return nil, nil, fmt.Errorf("compute withdrawals root: %w", err)
+		}
+		eph := &deneb.ExecutionPayloadHeader{
+			ParentHash:       payload.ParentHash,
+			FeeRecipient:     payload.FeeRecipient,
+			StateRoot:        payload.StateRoot,
+			ReceiptsRoot:     payload.ReceiptsRoot,
+			LogsBloom:        payload.LogsBloom,
+			PrevRandao:       payload.PrevRandao,
+			BlockNumber:      payload.BlockNumber,
+			GasLimit:         payload.GasLimit,
+			GasUsed:          payload.GasUsed,
+			Timestamp:        payload.Timestamp,
+			ExtraData:        payload.ExtraData,
+			BaseFeePerGas:    payload.BaseFeePerGas,
+			BlockHash:        payload.BlockHash,
+			TransactionsRoot: txRoot,
+			WithdrawalsRoot:  wRoot,
+			BlobGasUsed:      payload.BlobGasUsed,
+			ExcessBlobGas:    payload.ExcessBlobGas,
+		}
+
+		bb := &apiv1electra.BlindedBeaconBlock{
+			Slot:          p.Electra.Block.Slot,
+			ProposerIndex: p.Electra.Block.ProposerIndex,
+			ParentRoot:    p.Electra.Block.ParentRoot,
+			StateRoot:     p.Electra.Block.StateRoot,
+			Body: &apiv1electra.BlindedBeaconBlockBody{
+				RANDAOReveal:           p.Electra.Block.Body.RANDAOReveal,
+				ETH1Data:               p.Electra.Block.Body.ETH1Data,
+				Graffiti:               p.Electra.Block.Body.Graffiti,
+				ProposerSlashings:      p.Electra.Block.Body.ProposerSlashings,
+				AttesterSlashings:      p.Electra.Block.Body.AttesterSlashings,
+				Attestations:           p.Electra.Block.Body.Attestations,
+				Deposits:               p.Electra.Block.Body.Deposits,
+				VoluntaryExits:         p.Electra.Block.Body.VoluntaryExits,
+				SyncAggregate:          p.Electra.Block.Body.SyncAggregate,
+				ExecutionPayloadHeader: eph,
+				BLSToExecutionChanges:  p.Electra.Block.Body.BLSToExecutionChanges,
+				BlobKZGCommitments:     p.Electra.Block.Body.BlobKZGCommitments,
+				ExecutionRequests:      p.Electra.Block.Body.ExecutionRequests,
+			},
+		}
+
+		return &api.VersionedProposal{Version: p.Version, Blinded: true, ElectraBlinded: bb}, bb, nil
+
+	case spec.DataVersionFulu:
+		// Fulu reuses Electra block structures in this codebase.
```

Comment:
Perhaps having this comment above `bb := &apiv1electra.BlindedBeaconBlock{` would be clearer as we don't reference electra on the line below